File: Exercises_C++/2313.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int a, b, c;
    cin >> a >> b >> c;

    if ((a + b <= c) || (b + c <= a) || (a + c <= b)) {
        cout << "Invalido" << endl;
    } else {
        bool isEquilatero = (a == b && b == c);
        bool isIsoceles = (a == b || b == c || a == c);
        bool isRetangulo = (a * a == b * b + c * c) || (b * b == a * a + c * c) || (c * c == a * a + b * b);

        if (isEquilatero) {
            cout << "Valido-Equilatero" << endl;
        } else if (isIsoceles) {
            cout << "Valido-Isoceles" << endl;
        } else {
            cout << "Valido-Escaleno" << endl;
        }

        if (isRetangulo) {
            cout << "Retangulo: S" << endl;
        } else {
            cout << "Retangulo: N" << endl;
        }
    }

    return 0;
}

File: Exercises_Python/2313.py


a,b,c = map(int, input().split())
a2 = a * a
b2 = b * b
c2 = c * c

if (a < b + c) and (b < a + c) and (c < a + b):
    if (a == b) and (b == c):
        print('Valido-Equilatero')
        if (a2 == b2 + c2) or (b2 == a2 + c2) or (c2 == a2 + b2):
            print('Retangulo: S')
        else:
            print('Retangulo: N')
    elif (a != b) and (a != c) and (b != c):
        print('Valido-Escaleno')
        if (a2 == b2 + c2) or (b2 == a2 + c2) or (c2 == a2 + b2):
            print('Retangulo: S')
        else:
            print('Retangulo: N')
    else:
        print('Valido-Isoceles')
        if (a2 == b2 + c2) or (b2 == a2 + c2) or (c2 == a2 + b2):
            print('Retangulo: S')
        else:
            print('Retangulo: N')
else:
    print('Invalido')

